bookkeeper-server/src/main/java/org/apache/bookkeeper/bookie/EntryLogger.java

```diff
@@ -416,8 +423,10 @@ File getCurCompactionLogFile() {
     protected void initialize() throws IOException {
         // Register listener for disk full notifications.
         ledgerDirsManager.addLedgerDirsListener(getLedgerDirsListener());
-        // create a new log to write
-        createNewLog();
+
+        if (ledgerDirsManager.hasWritableLedgerDirs()) {
+            createNewLog();
```

Comment:
I know that. Your change involves a refactor which take time to get in, and probably we can't take it as 4.7.0 release. This change I am doing here is fixing a bug for 4.7.0 release. I don't think it is needed to couple these two changes together.

Reply:
I'm not sure about 4.7.0 release/timelines. But this PR - https://github.com/apache/bookkeeper/pull/1281 shouldn't take more time considering we agreed upon new interface. But anyhow with my change since I'm removing intialize method, the change you made here will be removed as well and since initialize method is removed, Bookie would be able to start with are without any writableLedgerDirs, as this change intend to be.

Reply:
@reddycharan - I am starting the release 4.7.0 release this week. I don't think #1281 is able to make it to 4.7.0 since it is a bit *risky*. The change here is much simpler just to defer log creation.